packages/surveys/src/components/general/Survey.tsx

```diff
@@ -174,6 +175,11 @@ export function Survey({
         }
       }
     }
+    if (startAtQuestionId) {
+      if (isFormPrefilling && currIdxTemp === 0) return startAtQuestionId;
+      const startAtQuestionIdx = questions.findIndex((q) => q.id === startAtQuestionId);
+      if (startAtQuestionIdx > currIdxTemp + 1) return questions[startAtQuestionIdx].id;
+    }
```

Comment:
can you please add a few comments here? Why is this condition required, or is there any other explanation?